src/main/java/com/coreos/jetcd/EtcdAuth.java

```diff
@@ -0,0 +1,68 @@
+package com.coreos.jetcd;
+
+import com.coreos.jetcd.api.*;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.protobuf.ByteString;
+
+/**
+ * Interface of auth talking to etcd
+ */
+public interface EtcdAuth {
+
+    // ***************
+    // Auth Manage
+    // ***************
+
+    ListenableFuture<AuthEnableResponse> authEnable();
+
+    ListenableFuture<AuthDisableResponse> authDisable();
+
+    // ***************
+    // User Manage
+    // ***************
+
+    ListenableFuture<AuthUserAddResponse> userAdd(ByteString name, ByteString passWord);
+
+    ListenableFuture<AuthUserDeleteResponse> userDelete(ByteString name);
+
+    ListenableFuture<AuthUserChangePasswordResponse> userChangePassword(ByteString name, ByteString passWord);
+
+    ListenableFuture<AuthUserGetResponse> userGet(ByteString name);
+
+    ListenableFuture<AuthUserListResponse> userList();
+
+    // ***************
+    // User Role Manage
+    // ***************
+
+    ListenableFuture<AuthUserGrantRoleResponse> userGrantRole(ByteString name, ByteString role);
+
+    ListenableFuture<AuthUserRevokeRoleResponse> userRevokeRole(ByteString name, ByteString role);
+
+    // ***************
+    // Role Manage
+    // ***************
+
+    ListenableFuture<AuthRoleAddResponse> roleAdd(ByteString name);
+
+    ListenableFuture<AuthRoleGrantPermissionResponse> roleGrantPermission(ByteString role, ByteString key,
+                                                                          ByteString rangeEnd, Permission.Type permType);
+
+    ListenableFuture<AuthRoleGetResponse> roleGet(ByteString role);
+
+    ListenableFuture<AuthRoleListResponse> roleList();
+
+    ListenableFuture<AuthRoleRevokePermissionResponse> roleRevokePermission(ByteString role, ByteString key,
+                                                                            ByteString rangeEnd);
+
+    ListenableFuture<AuthRoleDeleteResponse> roleDelete(ByteString role);
+
+
+    /**
+     * Authentication with user name and password, the service will return token if authentication succeeds
+     * @param name
+     * @param passWord
+     * @return AuthenticateResponse with token or error information
+     */
+    ListenableFuture<AuthenticateResponse> authenticate(ByteString name, ByteString passWord);
```

Comment:
let's remove authenticate for now. Ideally, we should call authenticate internally inside the client library and setup authenticated connection for users. Users do not need to do this themselves. Or they have to attach the token to each requests, which is tedious.